### DISCUSSION TITLE
Deleting selection

### DISCUSSION THREAD

**Discussion by Patrick-Poitras** [Fuco1/smartparens#1163] [Q&A]
Hello.

I'd like for the behavior of inserting a pair with a highlighted region to be replacing the region with the pair.

That is, if I have `this is a |thing` with `thing` highlighted, I'd like the insertion of "(" to result in `this is a ()`. Currently it gives `this is a (thing)` when sp-autowrap-region is t, and `this is a ()thing` when it is nil.

Delete-selection-mode is t.

**Comment by Fuco1** [comment]
This currently doesn't seem to be supported, but I guess it actually is the correct behaviour to replace the selection with a pair in your case of settings.

**Comment by Fuco1** [comment]
I have created an issue #1164 to track the fix.  I'll close this discussion.
